pilot/pkg/kube/inject/inject.go

```diff
@@ -775,6 +775,13 @@ func intoObject(sidecarTemplate string, valuesConfig string, meshconfig *meshcon
 		fmt.Fprintf(os.Stderr, "Skipping injection because %q has host networking enabled\n", metadata.Name) //nolint: errcheck
 		return out, nil
 	}
+	//skip injection for injected pods
+	for _, c := range podSpec.Containers {
```

Comment:
This is a weird/unsupported use case, but this would break a test we have https://github.com/istio/tools/blob/master/perf/load/pilot/templates/services.yaml. The test creates a pod with a container named istio-proxy as a hack to get a pod with just the istio-proxy after injection (no other container in the pod).

This is a major hack though so if we can't support it going forward that is ok

Reply:
skip injection only if the containers amounts in podSpec is greater than 1

Reply:
@howardjohn  Kubernetes does not allow using same container names within a pod.